src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonfwJavaProfile.java

```diff
@@ -0,0 +1,31 @@
+package com.devonfw.ide.sonarqube.common.impl;
+
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
+import org.sonar.api.utils.AnnotationUtils;
+import org.sonar.plugins.java.api.JavaCheck;
+import org.sonarsource.api.sonarlint.SonarLintSide;
+
+/**
+ * @author lniazman
+ *
+ */
+@SonarLintSide
+public class DevonfwJavaProfile implements BuiltInQualityProfilesDefinition {
+
+  @Override
+  public void define(Context context) {
+
+    NewBuiltInQualityProfile devonfwJava = context.createBuiltInQualityProfile("Devonfw Java", "java");
+    org.sonar.check.Rule ruleAnnotation;
+
+    for (Class<? extends JavaCheck> check : DevonSonarRegistrar.checkClasses()) {
+
+      ruleAnnotation = AnnotationUtils.getAnnotation(check, org.sonar.check.Rule.class);
+      devonfwJava.activateRule(DevonSonarDefinition.REPOSITORY_KEY, ruleAnnotation.key());
```

Comment:
So If I am not mistaken you are creating a Quality Profile that "only" contains the rules from this plugin.
It is great that you figured out how to add all rules of this plugin in a generic way so we do not have to maintain this.
However, we once spent weeks to walk through the rules of all common SonarQube plugins and decide which rules to enable/disable and which severities they should be on. That is the key value of the `devonfw Java` quality profile.
It is available here:
https://github.com/devonfw/ide-settings/blob/master/sonarqube/profiles/Devon-Java.xml
I heared that it can not be imported into a recent SonarQube so we might have quite some work ahead here and find a way to keep this in a maintainable form. But the `devonfw Java` Quality profile has to contain all Java related rules as SonarQube only supports to have one Quality Profile per Language and Project.